### PR TITLE
Add sign-in screen tests and auth UI ViewModel/state

### PR DESCRIPTION
**Pull request by ziyad-m97**
Description

This pull request introduces a complete authentication UI with a ViewModel and state management, and adds comprehensive UI and unit tests. 
The new AuthUIScreen aligns with the app’s visual identity and provides clear, responsive interactions for Microsoft Entra ID and Switch edu sign-in flows.

What issue does it solve?

It standardizes the sign-in experience, adds robust state handling for the authentication flow, and significantly increases test coverage for both UI and logic, improving reliability and maintainability.

What functionality does it add?

A new AuthUIScreen that:
Displays EPFL and Euler branding with a polished layout and subtle accents.
Offers primary (Microsoft) and secondary (Switch edu) sign-in actions.
New UI tests and ViewModel tests covering core behavior and edge cases.

Related Issues
`https://github.com/CS311-Team04/euler/issues/25`

Changes Made
UI
Added app/src/main/java/com/android/sample/sign_in/AuthUIScreen.kt with:
Consistent visual style, EPFL/Euler logos, and decorative accents.
Primary button for Microsoft Entra ID, secondary for Switch edu.

State & Logic
Added AuthUiState.kt modeling Idle, Loading(provider), SignedIn, and Error.
Added AuthViewModel.kt handling sign-in intents, preventing re-entry while loading, and transitioning after a delay.

Testing
Added app/src/androidTest/java/com/android/sample/screen/AuthUIScreenTest.kt:
Verifies presence of key elements, content descriptions, loading indicators, and button enabled states.
Added app/src/androidTest/java/com/android/sample/screen/AuthViewModelTest.kt:
Validates initial state, loading transitions, signed-in completion, and re-entry prevention.
Tests multiple interactions.

[x] Code adheres to project coding standards.
[x] New and updated tests have been written and verified.
[x] Code is free of known issues.
[x] Documentation/comments added where helpful (test tags, state transitions).

### CODE REVIEW

File: app/src/main/java/com/android/sample/sign_in/AuthUIScreen.kt [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+                        initialOffsetY = { it / 8 },
+                        animationSpec = tween(800, easing = FastOutSlowInEasing))) {
+              Surface(
+                  modifier =
+                      Modifier.padding(horizontal = 16.dp).fillMaxWidth().testTag(AuthTags.Card),
+                  tonalElevation = 8.dp,
+                  shadowElevation = 24.dp,
+                  color = Color.White.copy(alpha = 0.98f),
+                  shape = RoundedCornerShape(24.dp),
+                  border = BorderStroke(3.dp, Color(0x994B5563))) {
```
Comment (marcoff62) [inline app/src/main/java/com/android/sample/sign_in/AuthUIScreen.kt]: As for the rest of the UI, it could be useful to create a dedicated file for the colors and theme to scale properly later on with light and dark mode if ever needed.

File: app/src/main/java/com/android/sample/sign_in/AuthUIScreen.kt [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+    modifier: Modifier = Modifier
+) {
+  val backgroundColor = Color(0xFFC63F3F)
+
+  Box(
+      modifier = modifier.fillMaxSize().background(backgroundColor).testTag(AuthTags.Root),
+      contentAlignment = Alignment.Center) {
+        Accents()
+
+        AnimatedVisibility(
```
Comment (marcoff62) [inline app/src/main/java/com/android/sample/sign_in/AuthUIScreen.kt]: the authuistate right now includes an error state but no visual UI feedback for it. Maybe adding a small error message or any type of feedback would help the user know the authentication failed.
